src/extra/components/pulses.py

```diff
@@ -310,6 +317,58 @@ def is_constant_pattern(self):
             all([len(x) == 1 for x in pulse_ids.groupby(level=1).unique()])
         )
 
+    def is_interleaved_with(self, other: PulsePattern) -> Optional[bool]:
+        """Check whether pulses are interleaved with other pattern.
+
+        This method returns True if and only if it can be safely
+        determined the pulse pattern described by this component and the
```

Comment:
Nitpicking:
```suggestion
        This method returns `True` if and only if it can safely
        determine that the pulse pattern described by this component and the
```

Reply:
Thanks!